chapter_5/exercise_5_06/getline.c

```diff
@@ -1,23 +1,19 @@
 #include <stdio.h>
 
-void get_line(char *s);
 
-int main(void)
-{
-  char string[150] = "";
+char *getln(char *line);
 
-  get_line(string);
-  puts(string);
-
-  return 0;
+int main() {
+    char line[100];
+    
+    getln(line);
+    printf("%s", line);
+    
+    return 0;
 }
 
-void get_line(char *s)
-{
-  while ((*s = getchar()) != EOF && (*s != '\n'))
-  {
-    ++s;
-  }
-
-  *s = '\0';
+char *getln(char *line) {
+    while ((*line++ = getchar()) != 10);
```

Comment:
@kyopa It's true, EOF and '\n' have their equivalent as numbers in the end. However, the code has to be read by humans first, then by the computers (if only the computers have to read the code, then we can write binary). So, it'll be amazing to not use "magic numbers" like 10, or 0, especially where there is already a dedicated global constant available like EOF or '\n'.

As a practice magic numbers must be avoided, since the idea is to write code that is easy to read and performant, not only performant. Using magic numbers doesn't change the performance of the program, it's the same. However, it's harder to read, especially when the "magic numbers" are not something very common like in this context.